contracts/adapters/StakeWiseReinvestmentExtension.sol

```diff
@@ -0,0 +1,137 @@
+/*
+    Copyright 2022 Index Cooperative.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    SPDX-License-Identifier: Apache License, Version 2.0
+*/
+
+pragma solidity 0.6.10;
+pragma experimental ABIEncoderV2;
+
+import { Address } from "@openzeppelin/contracts/utils/Address.sol";
+import { SafeCast } from "@openzeppelin/contracts/utils/SafeCast.sol";
+
+import { BaseExtension } from "../lib/BaseExtension.sol";
+import { PreciseUnitMath } from "../lib/PreciseUnitMath.sol";
+import { IBaseManager } from "../interfaces/IBaseManager.sol";
+import { ISetToken } from "../interfaces/ISetToken.sol";
+import { IAirdropModule } from "../interfaces/IAirdropModule.sol";
+import { ITradeModule } from "../interfaces/ITradeModule.sol";
+
+/**
+ * @title StakeWiseReinvestmentExtension
+ * @author FlattestWhite
+ * 
+ * Smart contract that enables reinvesting the accrued rETH2 into a SetToken into sETH2.
+ */
+contract StakeWiseReinvestmentExtension is BaseExtension {
+    
+    using Address for address;
+    using SafeCast for int256;
+
+    /* ============ Constants ============= */
+
+    address public constant S_ETH2 = 0xFe2e637202056d30016725477c5da089Ab0A043A;
+    address public constant R_ETH2 = 0x20BC832ca081b91433ff6c17f85701B6e92486c5;
+
+    /* ========== Structs ================= */
+
+    struct ExecutionSettings {
```

Comment:
Maybe add short comment explaining this struct and each field ror maximum readability.